### ISSUE TITLE
Flickering text and more (Frequent GPS speed update)

### ISSUE THREAD

**Issue by Davi427** [osmandapp/OsmAnd#21288]
### Description

At times, items on the map flicker. This is distracting. 

### Steps to reproduce

I haven't found a specific action that triggers this behaviour, it just happens.

### Actual result

https://github.com/user-attachments/assets/bbf4ec5c-ec91-480b-8d7d-a4ffe221907b

https://github.com/user-attachments/assets/fc94cfd8-2b60-4c9e-a539-fa191dbd7bde

https://github.com/user-attachments/assets/91f23bb5-abc2-46b8-94e5-ce5f1c961bad

### Expected result

No flicker

### Your Environment (required)

**WARNING** Crash-Logs **MAY** contain information you deem sensitive.
Review this **CAREFULLY** before posting your issue!

```
OsmAnd Version: OsmAnd+ 4.9.3, Released: 2024-11-01
Android/iOS version: 14
Device model: FP5
Crash-Logs: ?
```

**Comment by vshcherb** [comment]
Looks really bad and this totally unexplainable yet. Let's see if we can reproduce it.

**Comment by yuriiurshuliak** [comment]
Unfortunately, I was unable to reproduce this bug. I attempted reproduction with a navigation simulation using a reset car profile on Gustava Zemgala gatve, as shown in your videos. No flickering of names, house numbers, or roads was observed.

For further investigation, it would be helpful if you could send an export of your car profile settings along with the exact start and finish coordinates of the route where this bug occurs.

To export your profile settings, go to Menu → Settings → Export. You can compress the file into a zip or email it directly to us at support@osmand.net.

Please also try our nightly version, as this issue may already be fixed there: [Nightly builds](https://osmand.net/docs/versions/nightly_versions#nightly-builds).

**Comment by Davi427** [comment]
> Unfortunately, I was unable to reproduce this bug. I attempted reproduction with a navigation simulation using a reset car profile on Gustava Zemgala gatve, as shown in your videos. No flickering of names, house numbers, or roads was observed.
> 
> For further investigation, it would be helpful if you could send an export of your car profile settings along with the exact start and finish coordinates of the route where this bug occurs.
> 
> To export your profile settings, go to Menu → Settings → Export. You can compress the file into a zip or email it directly to us at [support@osmand.net](mailto:support@osmand.net).
> 
> Please also try our nightly version, as this issue may already be fixed there: [Nightly builds](https://osmand.net/docs/versions/nightly_versions#nightly-builds).

Yeah today I did that. Downloaded nightly and Lithuanian maps, didn't change anything about the profile and ran two navigation apps at the same time.
Once I saw my regular osmand flicker I switched to the nightly showing similar behaviour.


[There was supposed to be a 7MB video showing the two apps here but the current network connection doesn't let me upload that. I'll add it later when I am able.]



Could it have something to do with the zoom? It seems somewhat similar to the behaviour I reported with the jittery sundown widget ( #21213 ) going back and forth between 1 hour and 59 minutes. I see the app doing that with several things. If that happens to the zoom level showing and then hiding these items could make sense, right?

**Comment by Davi427** [comment]
https://github.com/user-attachments/assets/88a3ea44-142d-4b70-a126-5825bf4c9f11

The screen record mentioned before.

**Comment by Davi427** [comment]
https://github.com/user-attachments/assets/b48017e2-79d4-4ff7-9b7d-8b6ddf2d9f86

This is what I was talking about before, the next turn widget going back and forth between yellow and grey.
And the sun widget doing the same. Back and forth between minutes.

**Comment by Davi427** [comment]
https://github.com/user-attachments/assets/8d23d416-3f89-49f8-8f44-04a6311cfcb9

Maybe this is another clue? 3D relief,  vertical exaggeration.

**Comment by vshcherb** [comment]
I think it's duplicate of issue with frequent gps update, if update happens too often than it causes zoom changes often and so on. I see that speed widget changes much more frequest than 1 FPS, so it causes other issues. 

We need to understand how to get these GPS so frequently so we can filter out them and smooth. Smoothing GPS signals is not supported feature for now, we have smoothing Kalman filter only for map rotation by compass

**Comment by Davi427** [comment]
Yeah that GPS flickering is very annoying and I assumed it's the phone, since I noticed it after switching. Unfortunately there is no way to change anything about it, so I'll have to wait for you to fix it.
Actually it's one reason I am so happy with the vehicle metrics and was asking about easier connection, because that speed widget is accurate and doesn't have this annoying jitter.

**Comment by Davi427** [comment]
I will do this.

I also disabled 3D relief and that reduces the flickering a lot.

**Comment by Zirochkabila** [comment]
OsmAnd~ 5.0.0#3853m, Released: 2024-11-30


https://github.com/user-attachments/assets/29ede46a-07ff-44d0-a576-a67c8094066b

**Comment by yuriiurshuliak** [comment]
<table style="width:100%">
  <tr>
<th><img src="https://github.com/user-attachments/assets/1aa217d0-8cb1-487b-9ecf-44557c30109c"  width="300" height="auto"/> </th>
 <th><img src="https://github.com/user-attachments/assets/194d6bdb-5727-410b-ab4b-4316633a1b9b"  width="300" height="auto"/> </th>
  </tr>
</table>

**Comment by yuriiurshuliak** [comment]
We have reproduced the bug under this conditions:  

1. **Steps to Reproduce**:  
   - Import the profile settings from the attached [Export_2024-12-02.zip](https://github.com/user-attachments/files/17979367/Export_2024-12-02.zip).  
   - Build a route with the following coordinates:  
     - **Start**: 48.20366° N, 14.25990° E  
     - **Finish**: 48.17153° N, 13.99172° E  
   - Load and enable the **Terrain map**.  
   - Start a navigation simulation using the **Realistic Simulation** option.  

2. **Observed Behavior**:  
   - During the simulation, flickering elements on the map, such as street names, are clearly visible.   
   - This behavior is **not reproduced** when the Terrain map is **not loaded or enabled**.  

OsmAnd~ 5.0.0#3860m, released: 2024-12-02

<table style="width:100%">
  </tr>
  <tr>
    <td> <video source src="https://github.com/user-attachments/assets/0ddfd9c1-88d5-48d5-8a63-112ab748afdb"  width="300" height="auto"/>   </td>
    <td><video source src="https://github.com/user-attachments/assets/7acbd757-6150-4880-b946-759479c04071"  width="300" height="auto"/>   </td>
  </tr>
</table>

**Comment by vshcherb** [comment]
Requires testing https://github.com/osmandapp/OsmAnd/commit/aa04771397bb9ef23bad24ed5977078ea3bf32d7 (but here is there is a change from 100 ms to 5 seconds for network location updates)

**Comment by vshcherb** [comment]
Also for passive gps provider is disabled for Android API - Location Source. Please give a try to nightly version

**Comment by vshcherb** [comment]
Please specify to investigate further: if it's Stock Android or Lineage OS and what's chip system it's running (I guess if it's FP5 - Qualcomm QCM6490 )

**Comment by vshcherb** [comment]
Label flickering is now fixed. Though here we could change 
To do: https://developers.google.com/android/reference/com/google/android/gms/location/LocationRequest.Builder#public-locationrequest.builder-setminupdateintervalmillis-long-minupdateintervalmillis - to 500 ms (2 HZ > 1 HZ) to avoid sporadic animation and other issues

**Comment by Zirochkabila** [comment]
# Testing 1
`OsmAnd~ 5.0.0#4116m, Дата випуску: 2025-01-13` - Fixed
